src/tests/Microsoft.PowerFx.Interpreter.Tests/DateTimeTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public class DateTimeTests
+    {
+        private static readonly bool IsLocalUTC = TimeZoneInfo.Local.BaseUtcOffset == TimeSpan.Zero;
+
+        [Fact]
+        public async void DateTimeUTCTests()
+        {
+            var engine = new RecalcEngine(
+               new PowerFxConfig(System.Globalization.CultureInfo.InvariantCulture));
+            var utcNow = DateTime.UtcNow;
+            var localNow = utcNow.ToLocalTime();
+
+            // myDateTime variable has system's current local time.
+            var symbolTable = new SymbolTable();
+            var symbolValues = new SymbolValues(symbolTable);
+            var myDateTimeSlot = symbolTable.AddVariable("myDateTime", FormulaType.DateTime);
+            symbolValues.Set(myDateTimeSlot, FormulaValue.New(localNow));
+            
+            // Creates Today's expression such that IsToday() can be true for local
+            // and false for UTC
+            var todayLocalDateTimeSlot = symbolTable.AddVariable("todayLocalDateTime", FormulaType.DateTime);
+            var localToday = DateTime.Today;
+            if (TimeZoneInfo.Local.BaseUtcOffset > TimeSpan.Zero)
```

Comment:
>TimeZoneInfo.Local.BaseUtcOffset > TimeSpan.Zero [](http://example.com/codeflow?start=16&length=48)

This expression worries me because it will behave differently in different timezones, may pass in some cases, fail in others. - and appear non-deterministic.  For example, Seattle, Paris, IST.

Reply:
Since we can set the TimeZone on the engine  (see SymbolExtensions.SetTimeZone) , can we set the timezone to a specific value and make this determinsitic? 

We should never be calling TimeZoneInfo.Local.

Reply:
Now setting local timezone to IST.